open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,595 @@
+
+        JSONObject validationResponse = new JSONObject();
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+
+        //Get the request payload from the ConsentManageData
+        if (!(request instanceof JSONObject)) {
+            log.error(ErrorConstants.PAYLOAD_FORMAT_ERROR);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                    ErrorConstants.PAYLOAD_FORMAT_ERROR, ErrorConstants.PATH_REQUEST_BODY);
+        }
+
+        JSONObject requestBody = (JSONObject) request;
+        //Check request body is valid and not empty
+        JSONObject dataValidationResult = ConsentManageUtil.validateInitiationDataBody(requestBody);
+
+        if (!(boolean) dataValidationResult.get(ConsentExtensionConstants.IS_VALID)) {
+            log.error(dataValidationResult.get(ConsentExtensionConstants.ERRORS));
+            return dataValidationResult;
+        }
+
+
+        //Check consent  initiation is valid and not empty
+        JSONObject initiationValidationResult = VRPConsentRequestValidator.validateConsentInitiation(requestBody);
+
+        if (!(boolean) initiationValidationResult.get(ConsentExtensionConstants.IS_VALID)) {
+            log.error(initiationValidationResult.get(ConsentExtensionConstants.ERRORS));
+            return initiationValidationResult;
+        }
+
+
+        JSONObject controlParameterValidationResult = VRPConsentRequestValidator.
+                validateConsentControlParameters(requestBody);
+
+        if (!(boolean) controlParameterValidationResult.get(ConsentExtensionConstants.IS_VALID)) {
+            log.error(controlParameterValidationResult.get(ConsentExtensionConstants.ERRORS));
+            return controlParameterValidationResult;
+        }
+
+
+        JSONObject riskValidationResult = VRPConsentRequestValidator.validateConsentRisk(requestBody);
+
+        if (!(boolean) riskValidationResult.get(ConsentExtensionConstants.IS_VALID)) {
+            log.error(riskValidationResult.get(ConsentExtensionConstants.ERRORS));
+            return riskValidationResult;
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Checks if the given Object is a JSONObject and the JSONObject is non-empty .
+     *
+     * @param value The Object to be validated.
+     * @return true if the object is a non-null and non-empty JSONObject.
+     */
+
+    public static boolean isValidJSONObject(Object value) {
+        return value instanceof JSONObject && !((JSONObject) value).isEmpty();
+    }
+
+
+    /**
+     * Checks if the given Object is a non-null and non-empty JSONObject.
+     *
+     * @param value The Object to be validated.
+     * @return true if the Object is a non-null and non-empty string representing a valid date-time.
+     */
+    public static boolean isValidDateTimeObject(Object value) {
```

Comment:
Behavior of the method and comment seems not matching

Reply:
Fixed in [3205f1b]